packages/utils/src/lib/reports/generate-md-reports-diff-utils.ts

```diff
@@ -0,0 +1,182 @@
+import { InlineText, MarkdownDocument, md } from 'build-md';
+import { ReportsDiff } from '@code-pushup/models';
+import { pluralize, pluralizeToken } from '../formatting';
+import { objectToEntries } from '../transform';
+import { DiffOutcome } from './types';
+
+// to prevent exceeding Markdown comment character limit
+const MAX_ROWS = 100;
+
+export function summarizeUnchanged(
+  token: string,
+  { changed, unchanged }: { changed: unknown[]; unchanged: unknown[] },
+): string {
+  return [
+    changed.length > 0
+      ? pluralizeToken(`other ${token}`, unchanged.length)
+      : `All of ${pluralizeToken(token, unchanged.length)}`,
+    unchanged.length === 1 ? 'is' : 'are',
+    'unchanged.',
+  ].join(' ');
+}
+
+export function summarizeDiffOutcomes(
+  outcomes: DiffOutcome[],
+  token: string,
+): string {
+  return objectToEntries(countDiffOutcomes(outcomes))
+    .filter(
+      (entry): entry is [Exclude<DiffOutcome, 'unchanged'>, number] =>
+        entry[0] !== 'unchanged' && entry[1] > 0,
+    )
+    .map(([outcome, count]): string => {
+      const formattedCount = `<strong>${count}</strong> ${pluralize(
+        token,
+        count,
+      )}`;
+      switch (outcome) {
+        case 'positive':
+          return `👍 ${formattedCount} improved`;
+        case 'negative':
+          return `👎 ${formattedCount} regressed`;
+        case 'mixed':
+          return `${formattedCount} changed without impacting score`;
+      }
+    })
+    .join(', ');
+}
+
+export function createGroupsOrAuditsDetails<T extends 'group' | 'audit'>(
+  token: T,
+  { changed, unchanged }: ReportsDiff[`${T}s`],
+  ...[columns, rows]: Parameters<MarkdownDocument['table']>
+): MarkdownDocument {
+  if (changed.length === 0) {
+    return new MarkdownDocument().paragraph(
+      summarizeUnchanged(token, { changed, unchanged }),
+    );
+  }
+  return new MarkdownDocument()
+    .table(columns, rows.slice(0, MAX_ROWS))
+    .paragraph(
+      changed.length > MAX_ROWS &&
+        md.italic(
+          `Only the ${MAX_ROWS} most affected ${pluralize(
```

Comment:
This would be cool to see in a snapshot

Reply:
Added a test case in 94fed1a2c374afeb687f4d95632ede06d84e1155.